### PR TITLE
Return a 1d-array for a single header selection 

### PR DESCRIPTION
**Pull request by alexanderkuvaev**
1. Now the following code
    ```python
    offset = gather["offset"]
    ```
    returns a 1d array instead of 2d. Old behavior can be obtained by always passing a list of headers similar to `pandas` convention:
    ```python
    offset_2d = gather[["offset"]]
    ```

2. Now `__setitem__` of trace containers acts identically to that of a `pd.DataFrame`. Previously the following code worked:
    ```python
    survey[["my_header"]] = np.arange(survey.n_traces)
    ```
    Now, following the `pandas` convention, it raises an error. Two valid options are:
    ```python
    survey["my_header"] = np.arange(survey.n_traces)
    survey[["my_header"]] = np.arange(survey.n_traces).reshape(-1, 1)
    ```

3. Simplifies `IndexPart` instantiation a bit.

### CODE REVIEW

File: seismicpro/utils/general_utils.py [hunk 1]
```diff
@@ -48,25 +49,46 @@ def get_first_defined(*args):
...
-        col_values = df[col] if col in df.columns else df.index.get_level_values(col)
-        res.append(col_values.values)
+    for col in cols:
+        if col in df.columns:
+            col_values = df[col]
+        elif col in df.index.names:
+            col_values = df.index.get_level_values(col)
+        else:
+            raise KeyError(f"Unknown header {col}")
+        res.append(col_values.to_numpy())
```
Comment (anton-br) [inline seismicpro/utils/general_utils.py:78]: Consider check if any headers are missed before `for` to raise error for all missed headers, not just for the first one.

something like this:
```suggestion
    unknown_headers = set(cols) - set(df.columns)
    if unknown_headers:
        raise KeyError(f"Unknown headers: {', '.join(unknown_headers)}")
    for col in cols:
        if col in df.columns:
            col_values = df[col]
        elif col in df.index.names:
            col_values = df.index.get_level_values(col)
        res.append(col_values.to_numpy())
```

**Review by dpodvyaznikov — APPROVED** [review]

File: seismicpro/utils/general_utils.py [hunk 2]
```diff
@@ -48,25 +49,46 @@ def get_first_defined(*args):
...
+
+    if is_single_col:
+        return res[0]
     return np.column_stack(res)
 
 
+def set_cols(df, cols, values):
+    """Set given `values` to selected columns of the `df` DataFrame."""
+    cols, _ = _process_cols(df, cols)
+    values = pd.DataFrame(values).to_numpy()
```
Comment (nikita-klsh) [inline seismicpro/utils/general_utils.py:88]: Do you need this intermediate cast to `DataFrame`?

File: seismicpro/utils/general_utils.py (hunk 1 above)
Comment (alexanderkuvaev) [reply seismicpro/utils/general_utils.py:78]: The proposal does not handle `MultiIndex` columns which are used by `IndexPart`. The easiest proper way to check whether columns exist in a `DataFrame` requires iterative `in` check which I'd like to call once.

**Comment by alexanderkuvaev** [comment]
Update: now `__setitem__` of trace containers acts identically to that of a `pd.DataFrame`. Previously the following code worked.
```python
survey[["my_header"]] = np.arange(survey.n_traces)
```
Now, following the `pandas` convention, it raises an error. Two valid options are:
```python
survey["my_header"] = np.arange(survey.n_traces)
survey[["my_header"]] = np.arange(survey.n_traces).reshape(-1, 1)
```
This is a breaking change, requesting review again.

File: seismicpro/utils/general_utils.py (hunk 2 above)
Comment (alexanderkuvaev) [reply seismicpro/utils/general_utils.py]: See my comment below

File: seismicpro/containers.py [hunk 3]
```diff
@@ -69,7 +66,7 @@ def __getitem__(self, key):
         result : 2d np.ndarray
```
[the remark this answers is not in the document]
Comment (alexanderkuvaev) [reply seismicpro/containers.py]: Fixed

**Review by anton-br — APPROVED** [review]

**Review by nikita-klsh — APPROVED** [review]

**Pull request merged**